hasjob/views/helper.py

```diff
@@ -28,14 +28,38 @@
 gif1x1 = 'R0lGODlhAQABAJAAAP8AAAAAACH5BAUQAAAALAAAAAABAAEAAAICBAEAOw=='.decode('base64')
 
 
-@app.route('/_sniffle.gif')
+@app.route('/_sniffle', methods=['POST'])
 def sniffle():
-    return gif1x1, 200, {
-        'Content-Type': 'image/gif',
-        'Cache-Control': 'no-cache, no-store, must-revalidate',
-        'Pragma': 'no-cache',
-        'Expires': '0'
-    }
+    """
+    Load anon user:
+
+    1. If there's g.user and session['anon_user'], it loads that anon_user and tags with user=g.user, then removes anon
+    2. If there's no g.user and no session['anon_user'] and form submitted token matches session['au'], sets session['anon_user'] = 'test'
+    3. If there's no g.user and there is session['au'] != form['token'], loads g.anon_user
+    """
+    if not g.user:
+        if unicode(session['au']).startswith('test') and unicode(session['au']) == request.form.get('token'):
+            # This client sent us back our test cookie, so set a real value now
+            g.anon_user = AnonUser()
+            db.session.add(g.anon_user)
+            g.esession = EventSession.new_from_request(request)
+            g.esession.anon_user = g.anon_user
+            db.session.add(g.esession)
+            g.esession.load_from_cache(session['au'], UserEvent)
+            # We'll update session['au'] below after database commit
+        else:
+            # form submitted token doesn't match the already set session['au']
+            # XXX: We got a fake value? This shouldn't happen
+            g.event_data['anon_cookie_test'] = session['au']
+            session['au'] = u'test-' + unicode(uuid4())  # Try again
+            g.esession = EventSessionBase.new_from_request(request)
+        db.session.commit()
+
+    if g.anon_user:
+        session['au'] = g.anon_user.id
+        session.permanent = True
+
+    return Response("OK")
```

Comment:
Don’t reply with text. Return `{'status': 'ok'}` and wrap the view with `render_with(json=True)`.